superset/viz.py

```diff
@@ -290,10 +287,11 @@ def get_payload(self, query_obj=None):
         payload = self.get_df_payload(query_obj)
 
         df = payload.get('df')
-        if df is not None and len(df.index) == 0:
-            raise Exception('No data')
         if self.status != utils.QueryStatus.FAILED:
-            payload['data'] = self.get_data(df)
+            if df is None or df.empty:
+                payload['error'] = 'No data'
```

Comment:
It's more consistent to log the payload error as opposed to throwing an exception here.